### PR TITLE
add CI, convert to Arduino library

### PR DESCRIPTION
**Pull request by ianfixes**
I'm working on a [library](https://github.com/ianfixes/arduino_ci) that allows you to run unit tests and compiling examples on Travis CI.  I'm using your project to shake some of the bugs out of my implementation, so I'm contributing my edits in case you think they would be useful.

Since you don't have Travis CI enabled for this repository, you'll have to peek at mine to see the results:
https://travis-ci.org/ianfixes/Arduino-Queue.h/builds

### CODE REVIEW

File: library.properties [hunk 1]
```diff
@@ -0,0 +1,10 @@
+name=Queue
+version=0.0.1
+author=Steven de Salas <steven@desalasworks.com>
+maintainer=Steven de Salas <steven@desalasworks.com>
+sentence=Generic C++ circular queue
+paragraph=Works for any datatype and desired length
+category=Data Storage
+url=https://github.com/ifreecarve/arduino_ci/SampleProjects/DoSomething
```
Comment (per1234) [inline library.properties]: The URL should be https://github.com/sdesalas/Arduino-Queue.h

Comment (ianfixes) [reply library.properties]: Whoops! Good catch

**Comment by sdesalas** [comment]
Hi Ian, this just popped up on my radar. Is it still good to merge?

**Comment by sdesalas** [comment]
Hi @ianfixes, I see some memory leak errors in travis CI, but I'm not sure if its because of the tests (queue instances are not being disposed of), the ci framework or the actual library.

Greatful if you could take a look:

https://travis-ci.com/github/sdesalas/Arduino-Queue.h

**Comment by ianfixes** [comment]
Hahaha this is a blast from the past!  I thought the project had been abandoned or something, and put it out of my mind.

>  I though it was someone random trying to add build scripts to my project

At the time, that's exactly what was happening!  😂   I used your queue library as a guinea pig for being able to write my unit testing library, and contributed the result.

The error happening here looks like a segfault, which the library uses `libasan` to annotate.  If I had to guess, it could be that between the initial implementation of my library and the version that exists now are conflicting over an internal Queue class that I added.  I will have to give this a run locally and see what the latest version of arduino_ci comes up with.

**Comment by sdesalas** [comment]
No worries, I'll take a look at [the tests](https://github.com/sdesalas/Arduino-Queue.h/blob/master/test/queue-tests.cpp) too when i get a chance.

I have a feeling that setting `queue = null` might do the trick to release the memory.

**Pull request merged**